receiver/prometheusreceiver/internal/transaction.go

```diff
@@ -426,23 +506,36 @@ func (t *transaction) getMetrics() (pmetric.Metrics, error) {
 		rms := md.ResourceMetrics().AppendEmpty()
 		resource.CopyTo(rms.Resource())
 
-		for scope, mfs := range families {
+		for scopeKey, mfs := range families {
 			ils := rms.ScopeMetrics().AppendEmpty()
 			// If metrics don't include otel_scope_name or otel_scope_version
 			// labels, use the receiver name and version.
-			if scope == emptyScopeID {
+			if scopeKey == "" {
 				ils.Scope().SetName(mdata.ScopeName)
 				ils.Scope().SetVersion(t.buildInfo.Version)
 			} else {
+				scope, err := parseScopeID(scopeKey)
```

Comment:
if possible I'd want to do a hash lookup of the scope data rather than embedding it in the families map.